app/validators/upload_validator.rb

```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-require_dependency "file_helper"
+require "file_helper"
```

Comment:
Is this still required since all files in `lib` are autoloaded?

Reply:
At first I went for removing every instance of `require_dependency` as stated by the Rails docs but then it basically broke everything. So I simply used `require` instead of `require_dependency`. These calls are managed by Zeitwerk anyway. The problem we have is we have a bunch of stuff in `lib` that should not be autoloaded and is required _before_ the autoload mechanism is enabled (in `config/application.rb`, basically when the app boots before running initializers). Because one subtlety here is that autoloading only works _after_ initializers have run (that’s why in an initializer you have to use `#to_prepare` blocks when you want to use autoloaded classes like models for example). So it’s a bit maddening at some point :sweat_smile: